mlir/lib/Rewrite/CMakeLists.txt

```diff
@@ -1,3 +1,14 @@
+set(LLVM_OPTIONAL_SOURCES ByteCode.cpp)
+
+if(MLIR_ENABLE_PDL)
+set(MLIRRewritePDLDeps
+  MLIRPDLDialect
+  MLIRPDLInterpDialect
+  MLIRPDLToPDLInterp)
+else()
+  set(MLIRRewritePDLDeps)
+endif()
+
```

Comment:
Can we also handle `ByteCode.cpp` here instead of adding `#ifdef` in the code?

It could be its own library and MLIRRewrite could depend on it:

```
if(MLIR_ENABLE_PDL)
  add_mlir_library(MLIRRewriteBytecode
    ByteCode.cpp
  DEPENDS
    MLIRPDLDialect
    MLIRPDLInterpDialect
    MLIRPDLToPDLInterp)
else()
  add_mlir_library(MLIRRewriteBytecode)
endif()
```